src/assert.rs

```diff
@@ -128,6 +132,34 @@ impl Assert {
         self
     }
 
+    /// Sets environments variables for the command.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// extern crate assert_cli;
+    ///
+    /// assert_cli::Assert::command(&["printenv"])
+    ///     .with_env(&[("TEST_ENV", "OK")])
+    ///     .stdout().contains("TEST_ENV=OK")
+    ///     .execute()
+    ///     .unwrap();
+    ///
+    /// let env = assert_cli::Environment::empty()
+    ///     .insert("FOO", "BAR");
+    ///
+    /// assert_cli::Assert::command(&["printenv"])
+    ///     .with_env(&env)
+    ///     .stdout().is("FOO=BAR")
+    ///     .execute()
+    ///     .unwrap();
+    /// ```
```

Comment:
Great examples! Can you maybe add a third one that sets a var in the test's thread context and then shows that it's also available in the command output?